enhancements/per-node-configuration.md

```diff
@@ -0,0 +1,484 @@
+**As a cluster administrator**, I want nodes in different datacenters to connect to datacenter-local BGP routers so that cross-datacenter BGP traffic is minimized and datacenter-specific network policies can be enforced.
+**As a cluster administrator**, I want to configure different NIC names for different server vendors so that I can use vendor-specific interface naming without requiring identical hardware across the cluster.
+
+## Design Details
+
+### API Changes
+
+Add an optional `NodeSelector` field to the `UnderlaySpec` structure:
+
+```go
+type UnderlaySpec struct {
+    // NodeSelector specifies which nodes this Underlay applies to.
+    // If empty or not specified, applies to all nodes (backward compatible).
+    // Multiple Underlays with overlapping node selectors will be rejected by validation webhook.
+    // +optional
+    NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
+
+    // ASN is the local AS number to use for sessions with ToR switches.
+    // +kubebuilder:validation:Minimum=1
+    // +kubebuilder:validation:Maximum=4294967295
+    // +required
+    ASN uint32 `json:"asn,omitempty"`
+
+    // RouterIDCIDR is the IPv4 CIDR for router ID assignment on matching nodes.
+    // +kubebuilder:default="10.0.0.0/24"
+    // +optional
+    RouterIDCIDR string `json:"routeridcidr,omitempty"`
+
+    // Neighbors is the list of BGP neighbors for matching nodes.
+    // +kubebuilder:validation:MinItems=1
+    Neighbors []Neighbor `json:"neighbors,omitempty"`
+
+    // Nics is the list of physical NICs on matching nodes.
+    // +kubebuilder:validation:items:Pattern=`^[a-zA-Z][a-zA-Z0-9._-]*$`
+    // +kubebuilder:validation:items:MaxLength=15
+    Nics []string `json:"nics,omitempty"`
+
+    EVPN *EVPNConfig `json:"evpn,omitempty"`
+}
+```
+
+### Example Configurations
+
+#### Multi-Rack Configuration
+
+Different racks connect to different ToR switches:
+
+```yaml
+# Rack 1 nodes connect to ToR 1
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: Underlay
+metadata:
+  name: underlay-rack-1
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      topology.kubernetes.io/rack: rack-1
+  asn: 64512
+  evpn:
+    vtepcidr: 100.65.1.0/24
+  nics:
+    - toswitch
+  neighbors:
+    - asn: 64500
+      address: 192.168.1.254  # ToR switch for rack 1
+      bfd:
+        receiveInterval: 300
+        transmitInterval: 300
+---
+# Rack 2 nodes connect to ToR 2
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: Underlay
+metadata:
+  name: underlay-rack-2
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      topology.kubernetes.io/rack: rack-2
+  asn: 64512
+  evpn:
+    vtepcidr: 100.65.2.0/24
+  nics:
+    - toswitch
+  neighbors:
+    - asn: 64500
+      address: 192.168.2.254  # ToR switch for rack 2
+      bfd:
+        receiveInterval: 300
+        transmitInterval: 300
+```
+
+#### Multi-Datacenter Configuration
```

Comment:
how are these any different from the `Multi-Rack Configuration` ? I see you've dropped the bfd configurations - OK, might make sense - but, afaiu, the "big" difference is the labels used - which can be anything, as long as it matches the labels on the nodes.

Just making sure I get the API.